### PR TITLE
Fix ClusterScoped kube APIs not subscribing

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes regression from #2266

### CODE REVIEW

File: src/renderer/kube-object.store.ts [hunk 1]
```diff
@@ -281,21 +281,38 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
...
-        } else {
-          apis.forEach(api => {
-            this.loadedNamespaces.forEach((namespace) => {
-              this.watchNamespace(api, namespace, abortController);
-            });
-          });
-        }
-      })
-      .catch(noop); // ignore DOMExceptions
+    console.log([clusterScopedApis, namespaceScopedApis]);
```
Comment (jim-docker) [inline src/renderer/kube-object.store.ts]: debug?

File: src/renderer/kube-object.store.ts [hunk 2]
```diff
@@ -281,21 +281,38 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
...
+
+    for (const api of namespaceScopedApis) {
+      const store = apiManager.getStore(api);
+
+      // This waits for the context and namespaces to be ready or fails fast if the disposer is called
+      Promise.race([rejectPromiseBy(abortController.signal), Promise.all([store.contextReady, store.namespacesReady])])
```
Comment (jim-docker) [inline src/renderer/kube-object.store.ts:290]: It was also a bug that this race was performed only for the `this` store?

**Review by jim-docker — DISMISSED** [review]

File: src/renderer/kube-object.store.ts (hunk 2 above)
Comment (Nokel81) [reply src/renderer/kube-object.store.ts:290]: Yes I think so. Though I now that I think about it. Subscribing really should be reactive on the change set of selected namespaces (adding and removing the watches as the namespaces get selected and deselected).

But I think that is out of scope of this fix.

**Review by jakolehm — APPROVED** [review]

**Review by nevalla — APPROVED** [review]
LGTM. Having integration test addressing this would be nice too!

**Comment by Nokel81** [comment]
> LGTM. Having integration test addressing this would be nice too!

I'll make an issue about that since I agree.

**Pull request merged**